pkg/kubelet/lifecycle/predicate_test.go

```diff
@@ -249,6 +250,80 @@ func TestGeneralPredicates(t *testing.T) {
 			reasons: []PredicateFailureReason{&PredicateFailureError{nodeports.Name, nodeports.ErrReason}},
 			name:    "hostport conflict",
 		},
+		{
```

Comment:
Is there a way to add tests for the static pod case?

Reply:
Added.